Text/Megaparsec/Char/Lexer.hs

```diff
@@ -346,7 +346,7 @@ charLiteral = label "literal character" $ do
   r <- lookAhead (count' 1 10 anySingle)
   case listToMaybe (Char.readLitChar r) of
     Just (c, r') -> c <$ skipCount (length r - length r') anySingle
-    Nothing -> unexpected (Tokens (head r :| []))
+    Nothing -> unexpected (Tokens (NE.fromList (take 1 r)))
```

Comment:
Isn't a use of `NE.fromList` as bad as using `head`?  Both will throw the exception in the _library_, not in the _application_ code.
If all that _head is bad_ excitement should lead to improvement, then the new code should be `fromMaybe undefined $ NE.nonEmpty $ take 1 r` which will raise the error (via `undefined`) here, not in the library.

Reply:
Both `head` and `NE.fromList` use `HasCallStack`, so they will at least point to the code here. But yeah, I don't think there is any big advantage of using `NE.fromList` rather than `head` other than to avoid the warning (in case that is even desired).